CommonMark/Parser/InlineMethods.cs

```diff
@@ -27,7 +28,7 @@ internal static class InlineMethods
             p['_'] = HandleEmphasis;
             p['*'] = HandleEmphasis;
             p['['] = HandleLeftSquareBracket;
-            p[']'] = HandleRightSquareBracket;
+            p[']'] = subj => HandleRightSquareBracket(subj, placeholderBracket);
```

Comment:
ILSpy confirms this suspicion - in fact, the compiler even creates an anonymous singleton with two methods that wrap the calls to `HandleRightSquareBracket`.

I made a few test runs with your CommonMarkBenchmark for both versions of the code, and saw what might be a very slight tendency to be faster for the version modified based upon your suggestion. Hence, I will change this code for now.